src/token.ts

```diff
@@ -13,8 +13,18 @@ function isTokenRenewed(newToken: string | undefined, error: AxiosError) {
 }
 
 export function pluginToken(provider: TokenProvider): ZodiosPlugin {
+  let pendingRenew: Promise<string | undefined> | undefined;
+
   return {
     request: async (_, config) => {
+      if (pendingRenew) {
+        // Wait for any pending renew request
+        try {
+          await pendingRenew;
+        } catch (error) {
+          throw new axios.Cancel("Renew token request failed");
```

Comment:
Why rethrow another error and hide the original ? 
can you use ZodiosError (with a cause) so this way we at least retrow with a 'cause" that we can inspect.

Reply:
I'll try with the ZodiosError 👍🏼 

I was just thinking that throwing `axios.Cancel` would let you test the error with `axios.isCancel`. And in this case we're not making a request, we're cancelling it.